include/awkward/Identities.h

```diff
@@ -262,7 +262,7 @@ namespace awkward {
   ///    - {@link IdentitiesOf Identities32}, which is `IdentitiesOf<int32_t>`
   ///    - {@link IdentitiesOf Identities64}, which is `IdentitiesOf<int64_t>`
   template <typename T>
-  class EXPORT_SYMBOL IdentitiesOf: public Identities {
+  class EXPORT_SYMBOL EXPORT_TYPE IdentitiesOf: public Identities {
```

Comment:
I suppose, `EXPORT_SYMBOL` is a requirement strictly stronger than `EXPORT_TYPE`, so this won't be very useful.

Reply:
I wasn't able to find any information about the relationship between `__attribute__((visibility("default")))` and `__attribute__((type_visibility("default")))`. Or for that matter `type_visibility` vs `__type_visibility__`. I could determine experimentally that only Clang knew about `type_visibility`; GCC didn't recognize it. From there, I just tried every combination, as you could see.

I thought it might be relevant because all of the symbol visibility linker errors concern `typeinfo` symbols. Everything is hidden by default (pybind11 requires this), so I'm labeling just the things that _should_ be externally visible with a macro. I want this to include the `typeinfo` of concrete classes, but apparently the normal `visibility` doesn't do that. I had hoped `type_visibility` would fill in where `visibility` failed.

I've reverted all the `EXTERNAL_TYPE` declarations.